app/controllers/api/registrations_controller.rb

```diff
@@ -7,24 +7,14 @@ def create
     build_resource(sign_up_params)
     if resource.save
       save_detected_locale(resource)
-      if pending_membership_is_present?
+      if pending_membership_is_present? or pending_identity_is_present?
         sign_in resource
         flash[:notice] = t(:'devise.sessions.signed_in')
+        render json: { success: :ok, signed_in: true }
       else
         LoginTokenService.create(actor: resource, uri: URI::parse(request.referrer.to_s))
+        render json: { success: :ok }
       end
-      render json: { success: :ok }
-    else
-      render json: { errors: resource.errors }, status: 422
-    end
-  end
-
-  def oauth
```

Comment:
I did not realise the route was a callback route.

I just noticed that now that we need to ask them to tick the box (and confirm their Loomio name) when the signup by facebook, so they go through the normal signup form, which points to create. So I did not think we'd need oauth anymore,and we just and link the identity the way we link memberships with the pending_actions_helper.

Reply:
It isn't a callback route. It's a another create method just for when you sign in via a third party.

Because we now need a legal_accepted value when these people create accounts, they need to submit to the normal create route, and we'll consume the pending identity on reload.

This is the same thing that happens when people with pending memberships create accounts, and I think the pattern is quite good, and the two can be used together no problem.